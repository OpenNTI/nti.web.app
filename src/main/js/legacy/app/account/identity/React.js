const Ext = require('@nti/extjs');
const { getService } = require('@nti/web-client');
const { Identity } = require('@nti/web-profiles');
const LoginActions = require('internal/legacy/login/Actions');
require('internal/legacy/overrides/ReactHarness');

const AccountActions = require('../Actions');

module.exports = exports = Ext.define(
	'NextThought.app.account.identity.React',
	{
		extend: 'NextThought.ReactHarness',
		alias: 'widget.identity-react',
		component: Identity.Session,

		setTheme(scope, knockout) {},

		initComponent: function () {
			this.callParent(arguments);
			Object.assign(this.initialConfig, {
				addHistory: true,
				baseroute: '/app',
			});
			this.AccountActions = AccountActions.create();
			this.LoginActions = LoginActions.create();
		},

		async afterRender() {
			this.callParent(arguments);
			const service = await getService();
			const supportLinks = service.getSupportLinks();

			this.mon(Ext.getBody(), {
				click: this.onBodyClicks.bind(this, supportLinks),
			});
		},

		onBodyClicks(supportLinks, e) {
			let action = e.getTarget('a[data-action]')?.dataset.action;
			const link = e.getTarget('a[href]');
			const href = link?.getAttribute('href');

			// This is used to launch support logic the same way everywhere
			// we have a link that goes to the supportContact url. (so we can
			// just emit that url in the dom and not worry about coping logic)
			if (
				supportLinks.supportContact === href &&
				supportLinks.internalSupport
			) {
				action = action || 'support';
			}

			if (action) {
				if (call(this, `do-${action}`, link, e) !== false) {
					e.stopEvent();
				}
			}
		},

		showLink(target) {
			target.setAttribute('target', '_blank');
			return false;
		},

		'do-welcomeGuide'(target) {
			this.AccountActions.showWelcomePage(target.href);
		},

		'do-about': 'showLink',

		'do-privacy'(target) {
			this.AccountActions.showPrivacy(target);
		},

		'do-privacyForMinors'(target) {
			this.AccountActions.showChildrensPrivacy(target.href);
		},

		'do-terms'(target) {
			this.AccountActions.showTermsOfService(target.href);
		},

		'do-support'(target) {
			this.AccountActions.showContactUs();
		},

		'do-helpSite': 'showLink',

		'do-impersonate'() {
			this.LoginActions.handleImpersonate();
		},

		'do-logout'() {
			this.LoginActions.handleLogout();
		},
	}
);

function call(scope, action, ...args) {
	if (typeof action === 'string') {
		return call(scope, scope[action], ...args);
	}

	return action.apply(scope, args);
}
