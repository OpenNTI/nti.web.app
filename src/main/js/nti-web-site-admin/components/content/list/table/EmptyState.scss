@import '~@nti/style-common';

.users-table-container {
	.empty-state {
		font-size: 2rem;
		color: $tertiary-grey;
		text-align: center;
		margin-top: 1.75rem;
	}
}
