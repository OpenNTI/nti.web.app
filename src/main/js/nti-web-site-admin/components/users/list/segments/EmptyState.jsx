import { EmptyState as Empty, Text } from '@nti/web-core';
import { scoped } from '@nti/lib-locale';

import { CreateSegmentButton } from './CreateButton';

const t = scoped('nti-site-admin.users.list.segments.EmptyState', {
	message: 'There are no segments yet',
	callToAction: 'Go %(create)s some...',
});

const T = Text.Translator(t);

export function EmptyState() {
	return (
		<Empty
			header={<T localeKey="message" />}
			subHeader={
				<T
					localeKey="callToAction"
					with={{
						create: <CreateSegmentButton variant="link" />,
					}}
				/>
			}
		/>
	);
}
