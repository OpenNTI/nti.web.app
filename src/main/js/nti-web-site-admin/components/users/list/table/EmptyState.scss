@import '~@nti/style-common';

.users-table-container {
	.empty-state {
		font-size: 2rem;
		color: var(--tertiary-grey);
		text-align: center;
		margin-top: 1.75rem;
	}
}
