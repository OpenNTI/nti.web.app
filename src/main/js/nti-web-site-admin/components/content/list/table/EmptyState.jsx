import React from 'react';
import PropTypes from 'prop-types';
import {scoped} from '@nti/lib-locale';

const DEFAULT_TEXT = {
	emptyMessage: 'There are no users'
};

const t = scoped('nti-site-admin.users.list.table.EmptyState', DEFAULT_TEXT);

EmptyState.propTypes = {
	message: PropTypes.string
};

export default function EmptyState ({message}) {
	return (
		<div className="empty-state">{message || t('emptyMessage')}</div>
	);
}
