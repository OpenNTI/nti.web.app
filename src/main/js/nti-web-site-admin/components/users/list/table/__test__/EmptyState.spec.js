/* eslint-env jest */
import React from 'react';
import { create, act } from 'react-test-renderer';

import EmptyState from '../EmptyState';

describe('Site admin user table empty state test', () => {
	test('Test default wording', () => {
		let cmp;
		act(() => {
			cmp = create(<EmptyState />);
		});

		const tree = cmp.toJSON();

		// should just use the default label for empty component
		expect(tree).toMatchSnapshot();
		cmp.unmount();
	});

	test('Test custom wording', () => {
		let cmp;
		act(() => {
			cmp = create(<EmptyState message="this is custom text" />);
		});

		const tree = cmp.toJSON();

		// should show "this is custom text"
		expect(tree).toMatchSnapshot();
		cmp.unmount();
	});
});
