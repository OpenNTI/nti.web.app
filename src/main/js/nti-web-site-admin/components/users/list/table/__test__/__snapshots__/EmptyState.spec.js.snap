// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site admin user table empty state test Test custom wording 1`] = `
<div
  className="empty-state"
>
  this is custom text
</div>
`;

exports[`Site admin user table empty state test Test default wording 1`] = `
<div
  className="empty-state"
>
  There are no users
</div>
`;
